pkg/operatorstatus/monitor.go

```diff
@@ -0,0 +1,215 @@
+package operatorstatus
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+const (
+	// clusterOperatorName is the name of the ClusterOperator.
+	clusterOperatorName = "marketplace"
+
+	// minimumSyncs represents the minimum number of syncs Marketplace must see
+	// prior to reporting that it is available to the ClusterOperator.
+	minimumSyncs uint = 2
+
+	// degradedRatio is the ratio of failed transition to syncs that must be reached
+	// prior to reporting that the marketplace operator is  degraded.
+	degradedRatio = 0.7
+)
+
+var (
+	instance *statusMonitor
+	once     sync.Once
+)
+
+type statusMonitor struct {
+	coWriter     *writer
+	namespace    string
+	eventTracker eventTracker
+	eventCh      chan error
+}
+
+// initInstance initializes the statusMonitor singleton named instance.
+func initInstance() error {
+	// Get the watch namespace.
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return fmt.Errorf("failed to get watch namespace: %v", err)
+	}
+
+	// Create a new writer.
+	coWriter, err := newWriter()
+	if err != nil {
+		return fmt.Errorf("Error creating ClusterOpreator writer: %v", err)
+	}
+
+	// Initialize the instance singleton once.
+	once.Do(func() {
+		// init the instance singleton.
+		instance = &statusMonitor{
+			coWriter:     coWriter,
+			namespace:    namespace,
+			eventTracker: eventTracker{0, 0, 1000, 10, sync.Mutex{}},
```

Comment:
Initting this inline makes sense, but can we use the longform so it's clear what these values are? And can we comment on each of them so they aren't just magic numbers?